python/pyspark/sql/functions.py

```diff
@@ -1059,36 +1236,50 @@ def shiftRightUnsigned(col, numBits):
     return Column(jc)
 
 
-@since(1.6)
 def spark_partition_id():
     """A column for partition ID.
 
-    .. note:: This is indeterministic because it depends on data partitioning and task scheduling.
+    .. versionadded:: 1.6.0
 
+    Notes
+    -----
+    This is indeterministic because it depends on data partitioning and task scheduling.
+
+    Examples
+    --------
     >>> df.repartition(1).select(spark_partition_id().alias("pid")).collect()
     [Row(pid=0), Row(pid=0)]
     """
     sc = SparkContext._active_spark_context
     return Column(sc._jvm.functions.spark_partition_id())
 
 
-@since(1.5)
 def expr(str):
     """Parses the expression string into the column that it represents
 
+    .. versionadded:: 1.5.0
+
+    Examples
+    --------
     >>> df.select(expr("length(name)")).collect()
     [Row(length(name)=5), Row(length(name)=3)]
     """
     sc = SparkContext._active_spark_context
     return Column(sc._jvm.functions.expr(str))
 
 
-@since(1.4)
 def struct(*cols):
     """Creates a new struct column.
 
-    :param cols: list of column names (string) or list of :class:`Column` expressions
+    .. versionadded:: 1.4.0
+
+    Parameters
+    ----------
+    cols : list, set, str or :class:`Column`
+        column names or :class:`Column`\\s to contain in the output struct.
```

Comment:
There are a bit of trivial changes in the doc to make it more sense with the new format. One example is here.